weaver/utils.py

```diff
@@ -1870,12 +1870,12 @@ def _request_call(method, url, kwargs):
 
 
 @cache_region("request")
-def _request_cached(method, url, kwargs):
-    # type: (AnyRequestMethod, str, RequestCachingKeywords) -> Response
+def _request_cached(method, url, kwargs, session):
+    # type: (AnyRequestMethod, str, RequestCachingKeywords, Optional[requests.Session]) -> Response
```

Comment:
Not sure if the session object is serializable for the caching decorator. 

Would also avoid the extra parameter. Everything should be in `RequestCachingKeywords`.

Reply:
> Not sure if the session object is serializable for the caching decorator.

Local tests seem to show that it caches it ok.

> Everything should be in RequestCachingKeywords

Are you sure? These get passed directly to `session.request` in the underlying function and that doesn't have a `session` parameter. So we'd have to factor it out later on.